_overviews/tutorials/binary-compatibility-for-library-authors.md

```diff
@@ -0,0 +1,215 @@
+Semantic Versioning v2.0.0.
+
+### Recommended Versioning Scheme
+
+* If backward **binary compatibility** is broken, **major version number** must be increased
+* If backward **source compatibility** is broken, **minor version number** must be increased
+* A change in **patch version number** signals **no binary nor source incompatibility**. According to SemVer, patch versions should contain only bug fixes that fix incorrect behavior so major behavioral
+change in method/classes should result in a minor version bump.
+* When major version is `0`, a minor version bump **may contain both source and binary breakages**
+* Some libraries may take a harder stance on maintaining source compatibility, bumping the major version number for ANY source incompatibility even if they are binary compatible
+
+Some examples:
+
+* `v1.0.0 -> v2.0.0` is <span style="color: red">binary incompatible</span>. Cares needs to be taken to make sure no evicted versions are still in the `v1.x.x` range to avoid runtime errors
+* `v1.0.0 -> v1.1.0` is <span style="color: blue">binary compatible</span> and maybe source incompatible
+* `v1.0.0 -> v1.0.1` is <span style="color: blue">binary compatible</span> and source compatible
+* `v0.4.0 -> v0.5.0` is <span style="color: red">binary incompatible</span> and maybe source incompatible
+* `v0.4.0 -> v0.4.1` is <span style="color: blue">binary compatible</span> and source compatible
+
+Many libraries in the Scala ecosystem has adopted this versioning scheme. A few examples are [Akka](http://doc.akka.io/docs/akka/2.5/scala/common/binary-compatibility-rules.html),
+[Cats](https://github.com/typelevel/cats#binary-compatibility-and-versioning) and [Scala.js](https://www.scala-js.org/).
+
+If this version scheme is followed, reasoning about binary compatibility is now very simple:
+
+* Ensure major versions of all versions of a library in the dependency tree are the same
+* Pick the latest version and evict the rest (This is the default behavior of SBT).
+
+### Explanation
+
+Why do we use the major version number to signal binary incompatible releases?
+
+From our [example](#why-binary-compatibility-matters) above, we have learned two important lessons:
+
+* Binary incompatibility releases often leads to dependency hell, rendering your users unable to update any of their libraries without breaking their application
+* If a new library version is binary compatible but source incompatible, the user can simply fix the compile errors in their application and everything will work
+
+Therefore, **binary incompatible releases should be avoided if possible** and be more noticeable when they happen, warranting the use of the major version number. While source compatibility
+is also important, if they are minor breakages that do not require effort to fix, then it is best to let the major number signal just binary compatibility.
```

Comment:
<s>if they are minor breakages that do not require effort to fix</s>
  
The previous statement is true no matter whether that conditional statement is true or not. The scheme signals the most severe API change, in this case breaking binary compatibility, and users should be prepared for source incompatible changes too.

Reply:
I'm just gonna take out this sentence. I feel like I'm overexplaining at various places when the reader can simply figure it out later on their own